### PR TITLE
[OKU-13] Header component-created reusable header component

### PR DESCRIPTION
**Pull request by twentymurial33**
## Summary of changes

Create a reusable header component 
image

## Change Type
*Put an `x` in all the boxes that apply
- [] Bugfix (non-breaking change which fixes an issue)
- [X] Feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to change)

## Linear issue link
https://linear.app/oku-no-kumo/issue/OKU-13/header-component

## Before & after screenshots or video recordings (Before & After)
Screen Shot 2023-02-23 at 12 28 15 PM

### CODE REVIEW

**Comment by idiglove** [comment]
Hi Murial, pls check https://github.com/idiglove/oku-no-kumo-fe/blob/main/CONTRIBUTING.md
for guidelines on making a PR

File: project_app/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    display: flex;
+    position: absolute;
+    width: 1440px;
+    height: 204px;
+  }
+  h1 {
+    position: absolute;
+    width: 227.77px;
+    height: 34.34px;
+    margin-right: 80px;
```
Comment (idiglove) [inline project_app/components/Header.js:30]: hmm, why would have h1 have dimensions?

Comment (twentymurial33) [reply project_app/components/Header.js]: I initially thought we were pulling the code from figma but now I know better 👍

File: project_app/components/Header.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  );
+}
+
+const Container = styled.div`
+  h1 {
+    position: absolute;
+    width: 227.77px;
+    height: 34.34px;
+    left: 50px;
+    top: 50px;
```
Comment (idiglove) [inline project_app/components/Header.js:52]: let's separate styled comp files to another file

File: project_app/components/Header.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+import Card from "@mui/material/Card";
+
+function Header() {
+  return (
+    <>
+      <AppBar
+        style={{ height: "412px", backgroundColor: "#353535", width: "1440px" }}
+      >
+        <Container maxWidth="xl">
+          <h1 className="u-text--center">AIN'T BOARD</h1>
```
Comment (idiglove) [inline project_app/components/Header.js:17]: where is this class been used in CSS `u-text--center`?

**Review by idiglove — COMMENTED** [review]
I know this is still in draft, I did see something tho. 
I see that the Footer component work is done here too? I actually prefer leaner PRs, so next time let's separate the PRs by their tickets. But for now, add `[OKU-18]` on the PR title too

File: project_app/components/Header.js (hunk 2 above)
Comment (twentymurial33) [reply project_app/components/Header.js:52]: Acknowledged

File: project_app/components/Header.js (hunk 3 above)
Comment (twentymurial33) [reply project_app/components/Header.js]: Removed that

**Comment by twentymurial33** [comment]
> I know this is still in draft, I did see something tho. I see that the Footer component work is done here too? I actually prefer leaner PRs, so next time let's separate the PRs by their tickets. But for now, add `[OKU-18]` on the PR title too

I will have the footer in its own PR - I can remove that -thank you for calling it out

File: project_app/components/Header.js [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+            <Button>Login</Button>
+            <Button>Join Us!</Button>
+          </div>
+          <Image
+            src="/logo.svg"
+            width={227.77}
+            height={150}
+            left={50}
+            right={50}
+            alt="image"
```
Comment (idiglove) [inline project_app/components/Header.js:24]: alt should be "Ain't Board logo"

File: project_app/components/Header.js [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+import InputAdornment from "@material-ui/core/InputAdornment";
+import { Container, AppBar, Wrapper, Typography } from "./styled";
+import Image from "next/image";
+
+function Header() {
+  return (
+    <>
+      <AppBar>
+        <Container maxWidth="xl">
+          <div style={{ position: "absolute", marginLeft: "540px" }}>
```
Comment (idiglove) [inline project_app/components/Header.js:14]: put these styles in styled component

File: project_app/components/Header.js [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+              ),
+            }}
+          />
+          <div
+            style={{
+              display: "flex",
+              marginRight: "500px",
+              alignItems: "center",
+              backgroundColor: "#353535",
+            }}
```
Comment (idiglove) [inline project_app/components/Header.js:46]: move to styled components

File: project_app/tsconfig.json [hunk 7]
```diff
@@ -0,0 +1,29 @@
+{
+  "compilerOptions": {
+    "lib": [
+      "dom",
+      "dom.iterable",
+      "esnext"
+    ],
+    "allowJs": true,
+    "skipLibCheck": true,
+    "strict": false,
```
Comment (idiglove) [inline project_app/tsconfig.json:10]: strict should be true, otherwise Typescript will be quite useless

**Review by idiglove — CHANGES_REQUESTED** [review]
1. Search.jpg shouldnt be in rep
2. Cleanup other files/image that are not needed

Pls make the changes, if it will take longer, we may need to bring it to next Sprint

File: project_app/tsconfig.json (hunk 7 above)
Comment (twentymurial33) [reply project_app/tsconfig.json:10]: Updated

File: project_app/components/Header.js (hunk 6 above)
Comment (twentymurial33) [reply project_app/components/Header.js]: moved

File: project_app/components/Header.js [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+          <input
+            placeholder="Search Board Game"
+            InputProps={{
+              endAdornment: (
+                <Image
+                  src="/icons8.svg"
+                  width={227.77}
+                  height={150}
+                  alt="image"
+                />
```
Comment (idiglove) [inline project_app/components/Header.js:35]: no need for this, design it in css

File: project_app/components/Header.js [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+            </IconButton>
+            <IconButton label="FAQ">
+              <p>FAQ</p>
+            </IconButton>
+            <IconButton label="About Us">
+              <p>About Us</p>
+            </IconButton>
+            <IconButton label="Game night">
+              <p>Game Night</p>
+            </IconButton>
```
Comment (idiglove) [inline project_app/components/Header.js:51]: can we have this in a separate component?

File: project_app/components/Header.js [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+          <div
+            style={{
+              position: "absolute",
+              width: "629px",
+              height: "471px",
+              top: "215px",
+              left: "0",
+            }}
+          >
+            <Image src="/bella.png" fill="absolute" alt="image" />
```
Comment (idiglove) [inline project_app/components/Header.js:66]: lets have appropriate naming for the filenames

File: project_app/components/Header.js [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+          </div>
+
+          <div
+            style={{
+              position: "absolute",
+              width: "296px",
+              height: "250px",
+              left: "412.31px",
+              top: "422px",
```
Comment (idiglove) [inline project_app/components/Header.js:75]: can we have this css separated in styled comps?

File: project_app/components/styled.tsx [hunk 12]
```diff
@@ -0,0 +1,84 @@
+import styled from "styled-components";
+
+export const AppBar = styled.div`
+position: absolute;
+width: 1440px;
+height: 700px;
+left: -4px;
+top: 0px;
+background: #353535;
+`
```
Comment (idiglove) [inline project_app/components/styled.tsx:10]: lets apply prettier with husky so indentations will be handled

File: project_app/components/styled.tsx [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+text-align: center;
+text-transform: uppercase;
+color: white;
+z-index:1;
+`
+
+// export const Wrapper = styled.button` 
+  
+    
+// `;
```
Comment (idiglove) [inline project_app/components/styled.tsx:83]: lets remove comments

File: project_app/pages/HomePage.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import Layout from "../components/Layout";
+
+function HomePage() {
+  return (
+    <Layout>
+      <div>HomePage</div>
+    </Layout>
```
Comment (idiglove) [inline project_app/pages/HomePage.js:8]: this should be index.tsx

File: project_app/tsconfig.json (hunk 7 above)
Comment (idiglove) [inline project_app/tsconfig.json:10]: strict is still false

**Review by idiglove — CHANGES_REQUESTED** [review]
we're almost there! :) 

few things
1. pls remove Search.jpg and other unused svgs
2. pls put the app in the root url, not in /HomePage
3. make sure the trees are behind the buttons
4. the search bar doesnt look right
5. theres no space on join us on its right

File: project_app/components/styled.tsx (hunk 13 above)
Comment (twentymurial33) [reply project_app/components/styled.tsx:83]: done

File: project_app/tsconfig.json (hunk 7 above)
Comment (twentymurial33) [reply project_app/tsconfig.json:10]: Yes it was failing - I will ask you in Discord

File: project_app/pages/HomePage.js (hunk 14 above)
Comment (twentymurial33) [reply project_app/pages/HomePage.js:8]: Updated

File: project_app/components/Header.js (hunk 9 above)
Comment (twentymurial33) [reply project_app/components/Header.js:51]: Created buttongroup component

File: project_app/components/Header.js (hunk 10 above)
Comment (twentymurial33) [reply project_app/components/Header.js:66]: updated filenames

File: project_app/components/styled.tsx (hunk 12 above)
Comment (twentymurial33) [reply project_app/components/styled.tsx:10]: installed

Comment (idiglove) [reply project_app/components/styled.tsx:10]: i dont see anywhere in package.json or a .husky folder

**Comment by twentymurial33** [comment]
@idiglove updated per new designs

File: project_app/components/ButtonGroup.js [hunk 15]
```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import styled from "styled-components";
+
+import Button from "@mui/material/Button";
+
+function ButtonGroup() {
```
Comment (idiglove) [inline project_app/components/ButtonGroup.js:6]: the naming of this file is too generic. This should be NavBarButtons or something

File: project_app/components/ToolBar.js [hunk 16]
```diff
@@ -0,0 +1,64 @@
+import styled from "styled-components";
+import Image from "next/image";
+
+function ToolBar() {
+  return (
+    <Container>
+      <div className="search">
+        <div>
+          <input type="text" />
+          <h1>Search board game </h1>
```
Comment (idiglove) [inline project_app/components/ToolBar.js:10]: as i said in discord
```suggestion
          <input type="text" placeholder="Search board game" />
```
and make sure it's typeable, and the text is not white

**Review by idiglove — CHANGES_REQUESTED** [review]
It's gotten better exponentially. Just little more

File: project_app/components/ButtonGroup.js (hunk 15 above)
Comment (twentymurial33) [reply project_app/components/ButtonGroup.js:6]: Done

File: project_app/components/styled.tsx [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+    border-radius: 10px;
+}
+
+p{
+  font-family: 'Rubik';
+font-style: normal;
+font-weight: 500;
+font-size: 20px;
+line-height: 24px;
+letter-spacing: 0.025em;
```
[the remark this answers is not in the document]
Comment (twentymurial33) [reply project_app/components/styled.tsx]: ok

**Review by idiglove — APPROVED** [review]
lgtm 🏄🏽 
let's move little stuff to next sprint. good job over all in here :) 🎉

**Pull request merged**